### ISSUE TITLE
Allow custom data copy function

### ISSUE THREAD

**Issue by andreasrueedlinger** [swimlane/ngx-dnd#56]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => Search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Sorry, we will not be able to answer every support request.  Please consider other venues for support requests
```

**Current behavior**
Currently when copying a model value, `JSON.parse(JSON.stringify(dropElmModel))` is used to copy the value. If the model value has circular references and if you want to manipulate the model data before it is copied it would be nice to provide a function that replaces the JSON copy mechanism.

**Expected behavior**
`ngxDroppable` directive has a property `copyModelValue` where you can provide a custom function to copy a value.

**Comment by ErwinSanquer** [comment]
Any update regarding this feature ?
Great job btw !

**Comment by ErwinSanquer** [comment]
I try to change the value in the event (drag) but when it's drop it still keep the old value.
I did find a way to change the dropped Item, but definitly not that nice implementation.

```
<div class="ngx-dnd-container" (drop)="builderDrop($event)" ngxDroppable="oneway-target" [removeOnSpill]="true" [model]="droppedItems">
                        <div class="ngx-dnd-item" ngxDraggable="['oneway-target']" *ngFor="let item of droppedItems" [model]="item">{{item.name_application_field}}</div>
</div>
```

```
builderDrop(e: any) {
        const index = this.droppedItems.indexOf(e.value);
        this.droppedItems.splice(index, 1, new ApplicationField(new Date().getTime(), e.value.field_type,
            this.applicationId, e.value.field_type, {}));
}
```

Still should be nice to have some update feature or some explanation on how to achieve this properly.

**Comment by Hypercubed** [comment]
Yes, the `drop` event is how I am doing it now.  I would like to allow a custom method for copy but have yet decided the best ergonomics for this.  I've found that passing functions into components can be troublesome.

**Comment by ErwinSanquer** [comment]
Maybe allowing to use .map() could be the solution if you want to avoid passing a function ?

**Comment by Eve-Sama** [comment]
anything about this feature?

**Comment by Delagen** [comment]
Any updates?
